test/unit/type_check.test.ts

```diff
@@ -0,0 +1,12 @@
+import tsd from 'tsd';
+import { expect } from 'chai'
+
+describe('Exported Types', () => {
+    it('should be as expected', async () => {
+        const diagnostics = await tsd();
+        if (diagnostics.length !== 0) {
+            const messages =  diagnostics.map(d => `${d.fileName}:${d.line}:${d.column} - [${d.severity}]: ${d.message}`).join('\n')
+            expect.fail('\n' + messages)
+        }
```

Comment:
Unfortunately, this encapsulates all the typescript testing into one mocha test. I think this is the best we can do at the moment, but a future project might consider somehow getting more info and generating `it` blocks.

Reply:
good enough for now :)